doc/template-design

```diff
@@ -0,0 +1,6 @@
+Support Compare then Add function
+The name of the function are CmpThenAdd. The function will compare the value of the first parameter with the value of the second parameter.
+
+If the value of the first parameter is greater than the value of the second parameter, the function will return the value of the first parameter.
+If the value of the second parameter are greater than the value of the first parameter, the function will return the value of the second parameter.
+Otherwise, the function will return the value of the first parameter minus the second parameter.
```

Comment:
Should be `Otherwise, the function will return the value of the first parameter plus the second parameter.`